Avoid futures timing out when asking for services

- If Consul is not answering, we should not create more threads to get services but wait a bit. => later
- Also we should make the timeout configurable? => later
- [x] Can we make the consul code non-blocking? https://github.com/FRosner/cluster-broccoli/blob/0.7.0/server/src/main/scala/de/frosner/broccoli/services/ConsulService.scala#L114

```
2017-11-15 10:15:43.882 [ERROR] [1-worker-3] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:43.882 [ERROR] [1-worker-2] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:43.883 [ERROR] [1-worker-0] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:43.883 [ERROR] [1-worker-1] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:49.528 [ERROR] [1-worker-2] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:49.529 [ERROR] [1-worker-0] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:49.529 [ERROR] [1-worker-1] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:56.893 [ERROR] [1-worker-2] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:56.894 [ERROR] [1-worker-0] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:56.898 [ERROR] [1-worker-1] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
2017-11-15 10:15:56.899 [ERROR] [1-worker-3] d.f.broccoli.services.ConsulService - Failed to get service statuses from Consul: java.util.concurrent.TimeoutException: Futures timed out after [5 seconds]
```

--- a/server/src/main/resources/logback.xml
+++ b/server/src/main/resources/logback.xml
@@ -4,7 +4,7 @@
     <!-- encoders are assigned the type
          ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
     <encoder>
-      <pattern>%d{HH:mm:ss.SSS} %level %logger - %msg%n</pattern>
+      <pattern>%d{HH:mm:ss.SSS} %level %.10thread %logger - %msg%n</pattern>
     </encoder>
   </appender>
 

--- a/server/src/main/scala/de/frosner/broccoli/services/ConsulService.scala
+++ b/server/src/main/scala/de/frosner/broccoli/services/ConsulService.scala
@@ -129,7 +129,7 @@ class ConsulService @Inject()(configuration: Configuration, ws: WSClient)(implic
       }
       case Failure(throwable) =>
         consulReachable = false
-        log.error(s"Failed to get service statuses from Consul: ${throwable.toString}")
+        log.error(s"Failed to get service statuses for job '$jobId' from Consul: ${throwable.toString}")
         val unknownServices = serviceNames.map(unknownService)
         serviceStatuses = serviceStatuses.updated(jobId, unknownServices)
     }

--- a/server/src/main/scala/de/frosner/broccoli/services/ConsulService.scala
+++ b/server/src/main/scala/de/frosner/broccoli/services/ConsulService.scala
@@ -110,7 +110,7 @@ class ConsulService @Inject()(configuration: Configuration, ws: WSClient)(implic
           }
       }
     }
-    val serviceResponse = Try(Await.result(Future.sequence(serviceResponses), Duration(5, TimeUnit.SECONDS)))
+    val serviceResponse = Future.sequence(serviceResponses)
     def unknownService(name: String) = Service(
       name = name,
       protocol = "",
@@ -118,7 +118,7 @@ class ConsulService @Inject()(configuration: Configuration, ws: WSClient)(implic
       port = 80,
       status = ServiceStatus.Unknown
     )
-    serviceResponse match {
+    serviceResponse.onComplete {
       case Success(services) => {
         val healthyOrUnhealthyServices = services.flatten.map(service => (service.name, service)).toMap
         val allServices = serviceNames.map { name =>

